File: java/src/sdk/dataobjects/paypal/payflow/MerchantInfo.java
package paypal.payflow;

/**
 * Used for Merchant  related information.
 * <p>Use this class to set the customer related
 * information.</p>
 *
 *  .................
 * // inv is the Invoice object
 * .................
 *	// Set the Merchant Info details.
 *	MerchantInfo Merchant = New MerchantInfo();
 *	Merchant.MerchantCode = "MerchantXXXXX";
 *	Merchant.MerchantCity = "Anywhere";
 *	Inv.MerchantInfo = Merchant;
 * .................
 */
public final class MerchantInfo extends BaseRequestDataObject {

    private String merchDescr;
    private String merchSvc;
    private String merchantName;
    private String merchantStreet;
    private String merchantCity;
    private String merchantState;
    private String merchantCountryCode;
    private String merchantZip;
    private String merchantUrl;
    private String merchantVatNum;
    private String merchantInvoiceNum;
    private String merchantLocationId;
    private String merchantId;
    private String merchantContactInfo;

    /**
     * Constructor
     * <p>Use this class to set the Merchant related information. Used for Soft Descriptor.</p>
     */
    public MerchantInfo() {
    }

    protected void generateRequest() {
        try {
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTNAME, merchantName));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTSTREET, merchantStreet));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTCITY, merchantCity));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTSTATE, merchantState));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTZIP, merchantZip));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTCOUNTRYCODE, merchantCountryCode));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTURL, merchantUrl));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTVATNUM, merchantVatNum));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTINVOICENUM, merchantInvoiceNum));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTLOCATIONID, merchantLocationId));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTID, merchantId));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHANTCONTACTINFO, merchantContactInfo));

            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHDESCR, merchDescr));
            getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MERCHSVC, merchSvc));
        } catch (Exception ex) {
            ErrorObject err = new ErrorObject(PayflowConstants.SEVERITY_FATAL, "", ex.toString());
            if (getContext() != null) {
                getContext().addError(err);
            }
        }
    }

    /**
     * Gets the Merchant's Name.
     *
     * @return merchantName String
     *  <p>Maps to Payflow Parameter: MERCHANTNAME</p>
     */
    public String getMerchantName() {
        return merchantName;
    }

    /**
     * Sets the Merchant's Name.
     * <p>Merchant/Business Name</P>
     *
     * @param merchantName String
     *  <p>Maps to Payflow Parameter: MERCHANTNAME</p>
     */
    public void setMerchantName(String merchantName) {
        this.merchantName = merchantName;
    }

    /**
     * Gets the Merchant's Street.
     *
     * @return merchantStreet String
     *  <p>Maps to Payflow Parameter: MERCHANTSTREET</p>
     */
    public String getmerchantStreet() {
        return merchantStreet;
    }

    /**
     * Sets the Merchant's Street.
     *
     * @param merchantStreet String
     *  <p>Maps to Payflow Parameter: MERCHANTSTREET</p>
     */
    public void setMerchantStreet(String merchantStreet) {
        this.merchantStreet = merchantStreet;
    }

    /**
     * Gets the Merchant's City.
     *
     * @return merchantCity String
     *  <p>Maps to Payflow Parameter: MERCHANTCITY</p>
     */
    public String getMerchantCity() {
        return merchantCity;
    }

    /**
     * Sets the Merchant's City.
     *
     * @param merchantCity String
     *  <p>Maps to Payflow Parameter: MERCHANTCITY</p>
     */
    public void setMerchantCity(String merchantCity) {
        this.merchantCity = merchantCity;
    }

    /**
     * Gets the Merchant's State.
     *
     * @return merchantState String
     *  <p>Maps to Payflow Parameter: MERCHANTSTATE   </p>
     */
    public String getMerchantState() {
        return merchantState;
    }

    /**
     * Sets the Merchant's State.
     *
     * @param merchantState String
     *  <p>Maps to Payflow Parameter: MERCHANTSTATE</p>
     */
    public void setMerchantState(String merchantState) {
        this.merchantState = merchantState;
    }

    /**
     * Gets the Merchant's Zip.
     *
     * @return merchantZip String
     *  <p>Maps to Payflow Parameter: MERCHANTZIP</p>
     */
    public String getMerchantZip() {
        return merchantZip;
    }

    /**
     * Sets the Merchant's Zip.
     *
     * @param merchantZip String
     *  <p>Maps to Payflow Parameter: MERCHANTZIP</p>
     */
    public void setMerchantZip(String merchantZip) {
        this.merchantZip = merchantZip;
    }

    /**
     * Gets the Merchant Country Code.
     *
     * @return merchantCountyCode String
     *  <p>Maps to Payflow Parameter: MERCHANTCOUNTRYCODE</p>
     */
    public String getMerchantCountryCode() {
        return merchantCountryCode;
    }

    /**
     * Sets the Merchant Country Code.
     * @param merchantCountryCode String
     *  <p>Maps to Payflow Parameter: MERCHANTCOUNTRYCODE</p>
     */
    public void setMerchantCountryCode(String merchantCountryCode) {
        this.merchantCountryCode = merchantCountryCode;
    }

    /**
     * Gets the Merchant Url.
     *
     * @return merchantUrl String
     *  <p>Maps to Payflow Parameter: MERCHANTURL</p>
     */
    public String getMerchantUrl() {
        return merchantUrl;
    }

    /**
     * Sets the Merchant Url.
     * @param merchantUrl String
     *  <p>Maps to Payflow Parameter: MERCHANTURL</p>
     */
    public void setMerchantUrl(String merchantUrl) {
        this.merchantUrl = merchantUrl;
    }

    /**
     * Gets the Merchant VAT number.
     *
     * @return merchantVatNum String
     *  <p>Maps to Payflow Parameter: MERCHANTVATNUM</p>
     */
    public String getMerchantVatNum() {
        return merchantVatNum;
    }

    /**
     * Sets the Merchant VAT number.
     * @param merchantVatNum String
     *  <p>Maps to Payflow Parameter: MERCHANTVATNUM</p>
     */
    public void setMerchantVatNum(String merchantVatNum) {
        this.merchantVatNum = merchantVatNum;
    }

    /**
     * Gets the Merchant Invoice number.
     *
     * @return merchantInvoiceNum String
     *  <p>Maps to Payflow Parameter: MERCHANTINVOICENUM</p>
     */
    public String getMerchantInvoiceNum() {
        return merchantInvoiceNum;
    }
    /**
     * Sets the Merchant Invoice number.
     * @param merchantInvoiceNum String
     *  <p>Maps to Payflow Parameter: MERCHANTINVOICENUM</p>
     */
    public void setMerchantInvoiceNum(String merchantInvoiceNum) {
        this.merchantInvoiceNum = merchantInvoiceNum;
    }

    /**
     * Gets the Merchant Location Id.
     * Merchant assigned store or location number.
     * @return - String
     *  <p>Maps to Payflow Parameter: MERCHANTLOCATIONID</p>
     */
    public String getMerchantLocationId() {
        return merchantLocationId;
    }
    /**
     * Sets the Merchant Location Id.
     * @param merchantLocationId String
     *  <p>Maps to Payflow Parameter: MERCHANTLOCATIONID</p>
     */
    public void setMerchantLocationId(String merchantLocationId) {
        this.merchantLocationId = merchantLocationId;
    }

    /**
     * Gets the Merchant Id.
     * Processor assigned Id.
     * @return - String
     *  <p>Maps to Payflow Parameter: MERCHANTID
     */
    public String getMerchantId() {
        return merchantId;
    }
    /**
     * Sets the Merchant Id.
     * @param merchantId String
     *  <p>Maps to Payflow Parameter: MERCHANTID</p>
     */
    public void setMerchantId(String merchantId) {
        this.merchantId = merchantId;
    }

    /**
     * Gets the Merchant Contact Information.
     * Merchants telephone, URl or email.
     * @return - String
     *  <p>Maps to Payflow Parameter: MERCHANTCONTACTINFO
     */
    public String getMerchantContactInfo() {
        return merchantContactInfo;
    }
    /**
     * Sets the Merchant Contact Information.
     * @param merchantContactInfo String
     *  <p>Maps to Payflow Parameter: MERCHANTCONTANCTINFO</p>
     */
    public void setMerchantContactInfoId(String merchantContactInfo) {
        this.merchantContactInfo = merchantContactInfo;
    }
















    /**
     * Gets the Merchant's description.
     *
     * @return merchDescr String
     *  <p>Maps to Payflow Parameter: MERCHDESCR</p>
     */
    public String getMerchDescr() {
        return merchDescr;
    }

    /**
     * Sets the Merchant's description.
     *
     * @param merchDescr String
     *  <p>Maps to Payflow Parameter: MERCHDESCR</p>
     */
    public void setMerchDescr(String merchDescr) {
        this.merchDescr = merchDescr;
    }

    /**
     * Gets the Merchant's contact information.
     *
     * @return merchSvc String
     *  <p>Maps to Payflow Parameter: MERCHSVC</p>
     */
    public String getMerchSvc() {
        return merchSvc;
    }

    /**
     * Sets the Merchant's contact information.
     *
     * @param merchSvc String
     *  <p>Maps to Payflow Parameter: MERCHSVC</p>
     */
    public void setMerchSvc(String merchSvc) {
        this.merchSvc = merchSvc;
    }
}

File: dotNET/PayflowProSDK/PayPal/Payments/DataObjects/MerchantInfo.cs
#region "Imports"

using System;
using PayPal.Payments.Common.Utility;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.DataObjects
{
    /// <summary>
    /// Used for Merchant related information.
    /// </summary>
    /// <remarks>Use this class to set the Merchant related 
    /// information.  Used for Soft Descriptors.  
    /// 
    /// Refer to the Payflow Gateway Developer's Guide for your processor
    /// for more information related to this fields.</remarks>
    /// <example>
    /// <code lang="C#" escaped="false">
    ///	.................
    ///	// Inv is the Invoice object
    ///	.................
    ///	// Set the Merchant Info details.
    ///	MerchantInfo Merchant = new MerchantInfo();
    ///	Merchant.MerchantName = "MerchantXXXXX";
    ///	Merchant.MerchantCity = "Anywhere";
    ///	Inv.MerchantInfo = Merchant;
    ///	.................
    ///	</code>
    /// <code lang="Visual Basic" escaped="false">
    ///	.................
    ///	' Inv is the Invoice object
    ///	.................
    ///	' Set the Merchant Info details.
    ///	Dim Merchant As MerchantInfo = new MerchantInfo
    ///	Merchant.MerchantName = "MerchantXXXXX"
    ///	Merchant.MerchantCity = "Anywhere"
    ///	Inv.MerchantInfo = Merchant
    ///	.................
    ///	</code>
    /// </example>
    public sealed class MerchantInfo : BaseRequestDataObject
    {
        #region "Member Variables"

        /// <summary>
        /// Merchant Description
        /// </summary>
        private String mMerchDescr;
        /// <summary>
        /// Merchant Telephone
        /// </summary>
        private String mMerchSvc;
        /// <summary>
        /// Holds Merchant Name
        /// </summary>
        private String mMerchantName;
        /// <summary>
        /// Holds Merchant Street
        /// </summary>
        private String mMerchantStreet;
        /// <summary>
        /// Holds Merchant City
        /// </summary>
        private String mMerchantCity;
        /// <summary>
        /// Holds Merchant State
        /// </summary>
        private String mMerchantState;
        /// <summary>
        /// Holds Merchant CountryCode
        /// </summary>
        private String mMerchantCountryCode;
        /// <summary>
        /// Holds Merchant Zip
        /// </summary>
        private String mMerchantZip;
        /// <summary>
        /// Holds Merchant Url
        /// </summary>
        private String mMerchantUrl;
        /// <summary>
        /// Holds Merchant VAT Number
        /// </summary>
        private String mMerchantVatNum;
        /// <summary>
        /// Holds Merchant Invoice Number
        /// </summary>
        private String mMerchantInvNum;
        /// <summary>
        /// MerrchantLocationId
        /// </summary>
        private String mMerchantLocationId;
        /// <summary>
        /// MerchantId
        /// </summary>
        private String mMerchantId;
        /// <summary>
        /// MerchantContactInfo
        /// </summary>
        private String mMerchantContactInfo;

        #endregion

        #region "Constructor"
        /// <summary>
        /// Constructor
        /// </summary>
        /// <remarks>
        /// Use this class to set the Soft merchant information which is detailed data about a merchant such
        /// as the merchant's name, business address, business location identifier, and contact information.
        /// </remarks>
        /// <example>
        /// <code lang="C#" escaped="false">
        ///	.................
        ///	// Inv is the Invoice object
        ///	.................
        ///	// Set the Merchant Info details.
        ///	MerchantInfo Merchant = New MerchantInfo();
        ///	Merchant.MerchantName = "MerchantXXXXX";
        ///	Merchant.MerchantCity = "Somewhere";
        ///	Inv.MerchantInfo = Merchant;
        ///	.................
        ///	</code>
        /// <code lang="Visual Basic" escaped="false">
        ///	.................
        ///	' Inv is the Invoice object
        ///	.................
        ///	' Set the Merchant Info details.
        ///	Dim Merchant As MerchantInfo = New MerchantInfo
        ///	Merchant.MerchantName = "MerchantXXXXX"
        ///	Merchant.MerchantCity = "Somewhere"
        ///	Inv.MerchantInfo = Merchant
        ///	.................
        ///	</code>
        /// </example>
        public MerchantInfo()
        {
        }
        #endregion

        #region "Properties"
     
        /// <summary>
        /// Gets, Sets  Merchant Name
        /// </summary>
        /// <remarks>
        /// <para>Name of Merchant</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTNAME</code>
        /// </remarks>
        public String MerchantName
        {
            get { return mMerchantName; }
            set { mMerchantName = value; }
        }
        /// <summary>
        /// Gets, Sets  Merchant Street
        /// </summary>
        /// <remarks>
        /// <para>Merchant's Street Address (Number and Street Name)</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTSTREET</code>
        /// </remarks>
        public String MerchantStreet
        {
            get { return mMerchantStreet; }
            set { mMerchantStreet = value; }
        }
        /// <summary>
        /// Gets, Sets  Merchant City
        /// </summary>
        /// <remarks>
        /// <para>Merchant's City</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTCITY</code>
        /// </remarks>
        public String MerchantCity
        {
            get { return mMerchantCity; }
            set { mMerchantCity = value; }
        }
        /// <summary>
        /// Gets, Sets  Merchant State
        /// </summary>
        /// <remarks>
        /// <para>Merchant's State</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTSTATE</code>
        /// </remarks>
        public String MerchantState
        {
            get { return mMerchantState; }
            set { mMerchantState = value; }
        }
        /// <summary>
        /// Gets, Sets  Merchant Country Code
        /// </summary>
        /// <remarks>
        /// <para>Merchant's Numeric Country Code.  Example: USA = 840</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTCOUNTRYCODE</code>
        /// </remarks>
        public String MerchantCountryCode
        {
            get { return mMerchantCountryCode; }
            set { mMerchantCountryCode = value; }
        }
        /// <summary>
        /// Gets, Sets  Merchant Zip
        /// </summary>
        /// <remarks>
        /// <para>Merchant's 5- to 9-digit ZIP (postal) code excluding
        /// spaces, dashes, and non-numeric characters.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTZIP</code>
        /// </remarks>
        public String MerchantZip
        {
            get { return mMerchantZip; }
            set { mMerchantZip = value; }
        }

        /// <summary>
        /// Gets, Sets  Merchant Url
        /// </summary>
        /// <remarks>
        /// <para>Merchant's website (URL)</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTURL</code>
        /// </remarks>
        public String MerchantUrl
        {
            get { return mMerchantUrl; }
            set { mMerchantUrl = value; }
        }

        /// <summary>
        /// Gets, Sets  Merchant VAT Number
        /// </summary>
        /// <remarks>
        /// <para>Merchant's VAT Number</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTVATNUM</code>
        /// </remarks>
        public String MerchantVatNum
        {
            get { return mMerchantVatNum; }
            set { mMerchantVatNum = value; }
        }

        /// <summary>
        /// Gets, Sets  Merchant Invoice Number
        /// </summary>
        /// <remarks>
        /// <para>Merchant's Invoice Number</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTINVNUM</code>
        /// </remarks>
        public String MerchantInvNum
        {
            get { return mMerchantInvNum; }
            set { mMerchantInvNum = value; }
        }

        		/// <summary>
        /// Gets, Sets the Merchant Location Id.
        /// </summary>
		/// Merchant assigned store or location number.
        /// <remarks>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTLOCATIONID</code>
        /// </remarks>
        public String MerchantLocationId
        {
            get { return mMerchantLocationId; }
            set { mMerchantLocationId = value; }
        }
		/// <summary>
        /// Gets, Sets the Merchant Id.
        /// </summary>
		/// Processor assigned number.
        /// <remarks>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTID</code>
        /// </remarks>
        public String MerchantId
        {
            get { return mMerchantId; }
            set { mMerchantId = value; }
        }
		/// <summary>
        /// Gets, Sets the Merchant Contact Information.
        /// </summary>
		/// Merchants telephone, URl or email.
        /// <remarks>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHANTCONTACTINFO</code>
        /// </remarks>
        public String MerchantContactInfo
        {
            get { return mMerchantContactInfo; }
            set { mMerchantContactInfo = value; }
        }

        /// <summary>
        /// Gets, Sets  MerchDescr
        /// </summary>
        /// <remarks>
        /// <para>Merchant's description.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHDESCR</code>
        /// </remarks>
        /// 
        public String MerchDescr
        {
            get { return mMerchDescr; }
            set { mMerchDescr = value; }
        }

        /// <summary>
        /// Gets, Sets  MerchSvc
        /// </summary>
        /// <remarks>
        /// <para>Merchant's contact number.</para>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>MERCHSVC</code>
        /// </remarks>
        public String MerchSvc
        {
            get { return mMerchSvc; }
            set { mMerchSvc = value; }
        }

        #endregion

        #region "Core functions"

        /// <summary>
        /// Generates the transaction request.
        /// </summary>
        internal override void GenerateRequest()
        {
            try
            {
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTNAME, mMerchantName));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTSTREET, mMerchantStreet));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTCITY, mMerchantCity));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTSTATE, mMerchantState));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTZIP, mMerchantZip));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTCOUNTRYCODE, mMerchantCountryCode));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTURL, mMerchantUrl));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTVATNUM, mMerchantVatNum));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTINVNUM, mMerchantInvNum));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTLOCATIONID, mMerchantLocationId));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTID, mMerchantId));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHANTCONTACTINFO, mMerchantContactInfo));

                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHDESCR, mMerchDescr));
                RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MERCHSVC, mMerchSvc));
            }
            catch (BaseException)
            {
                throw;
            }
            catch (Exception Ex)
            {
                DataObjectException DEx = new DataObjectException(Ex);
                throw DEx;
            }
        }
        #endregion
    }
}